packages/engine/src/xr/systems/XRSystem.ts

```diff
@@ -30,7 +31,15 @@ const startXRSession = async () => {
     dispatchLocal(EngineActions.xrSession() as any)
 
     // Current WebXRManager.getCamera() typedef is incorrect
-    ;(Engine.xrManager as any).getCamera().layers.enableAll()
+    // @ts-ignore
+    const cameras = Engine.xrManager.getCamera() as ArrayCamera
+    cameras.layers.enableAll()
+    cameras.cameras.forEach((camera) => {
```

Comment:
`cameras.cameras` array is empty at this point. The VR camera layers are also the reason that the avatar is not showing.
I made a fix by running the same code in a world receptor as it is running a few frames later. At that point, the camera array is already populated.


`  Engine.currentWorld.receptors.push((action: EngineActionType) => {
    switch (action.type) {
      case NetworkWorldAction.setXRMode.type:
        // Current WebXRManager.getCamera() typedef is incorrect
        // @ts-ignore
        const cameras = Engine.xrManager.getCamera() as ArrayCamera
        cameras.cameras.forEach((camera) => {
          camera.layers.disableAll()
          camera.layers.enable(ObjectLayers.Scene)
          camera.layers.enable(ObjectLayers.Avatar)
          camera.layers.enable(ObjectLayers.UI)
        })
        break
    }
  })`